R/init.R

```diff
@@ -113,14 +121,24 @@ if (interactive() &&
             )
           } else {
             obj <- env[[name]]
-            str <- capture_str(obj)[[1L]]
+
+            addr <- address(obj)
+            cobj <- cache[[name]]
+            if (is.null(cobj) || cobj$address != addr) {
+              cache[[name]] <- cobj <- list(
```

Comment:
Would using addresses as keys in the cache work here?
Variables might share the same address, but I'd expect them to be identical then (i.e. have the same size as well).

Reply:
Yes, it should work. I'm curious about if it is possible that the same address will be reused?

Reply:
```
> rm(x); gc(); x <- rnorm(1000); .Internal(inspect(x));                         
         used (Mb) gc trigger (Mb) max used (Mb)
Ncells 278099 14.9     655202   35   462665 24.8
Vcells 556656  4.3    8388608   64  1779488 13.6
@1f83740 14 REALSXP g0c7 [REF(1)] (len=1000, tl=0) -0.491836,-0.481699,-0.205983,-0.750808,1.92488,...

> rm(x); gc(); x <- rnorm(1001); .Internal(inspect(x));                         
         used (Mb) gc trigger (Mb) max used (Mb)
Ncells 278099 14.9     655202   35   462665 24.8
Vcells 556656  4.3    8388608   64  1779488 13.6
@1f83740 14 REALSXP g0c7 [REF(1)] (len=1001, tl=0) -1.16595,0.958288,1.88609,0.510728,0.0115951,...

```

Looks like the memory address could be reused.

Reply:
(How) does the function recognize that `x` has a different size in the second callback? (with variable name and address being equal)

Reply:
I update the mechanism so that when the symbol is not in cache or address changes or length changes, the object size will be re-calculated.